### PR TITLE
new package: ceu-lang

### PR DESCRIPTION
**Pull request by pvonmoradi**
Adds [ceu language.](http://www.ceu-lang.org/)

### CODE REVIEW

File: packages/ceu-lang/build.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+TERMUX_PKG_HOMEPAGE="https://github.com/ceu-lang/ceu"
+TERMUX_PKG_DESCRIPTION="The Structured Synchronous Reactive Programming Language Céu"
+TERMUX_PKG_LICENSE="MIT"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_SRCURL="https://github.com/ceu-lang/ceu.git"
+TERMUX_PKG_GIT_BRANCH="master"
+_COMMIT="5e0c8d3004ad98658ffe82823ad8303a8d371064"
+TERMUX_PKG_VERSION="2019.07.17"
+TERMUX_PKG_BUILD_IN_SRC=true
+TERMUX_PKG_BUILD_DEPENDS="lua53, lua-lpeg, liblua53"
```
Comment (Grimler91) [inline packages/ceu-lang/build.sh:10]: Is this based on test builds on device? For cross-compiling I would expect lua53 to be required in docker image ([setup-ubuntu.sh](https://github.com/termux/termux-packages/blob/master/scripts/setup-ubuntu.sh)), and liblua53 to be a runtime dependency, in `TERMUX_PKG_DEPENDS`

Comment (pvonmoradi) [reply packages/ceu-lang/build.sh]: The ceu compiler  is actually a `lua5.3` script. I tested ceu on target (aarch64) built with this: 
```
TERMUX_PKG_BUILD_DEPENDS="lua53, lua-lpeg"
TERMUX_PKG_DEPENDS="lua53, lua-lpeg, liblua53"
```

**Comment by pvonmoradi** [comment]
### Testing ceu-lang

`git clone https://github.com/ceu-lang/ceu` 

Running `make samples` on aarch64. Out of 5 test samples, one does not build due to pthread_cancel (`test-03.ceu`).

```
for i in samples/*.ceu; do                                              \
	echo;                                                               \
	echo -n "#####################################";                    \
	echo    "#####################################";                    \
	echo File: "$i -> /data/data/com.termux/files/usr/tmp/$(basename $i .ceu)";	                    \
	grep "#@" "$i" | cut -f2- -d" ";                                   \
	echo -n "#####################################";                    \
	echo    "#####################################";                    \
	echo -n "Press <enter> to start...";                                \
	read _;                                                             \
	echo ceu --pre --pre-input=$i --pre-args=\"-I./include\"           \
        --ceu --ceu-features-os=true --ceu-features-lua=true --ceu-features-thread=true --ceu-features-dynamic=true --ceu-err-unused=pass \
	    --env --env-types=env/types.h --env-threads=env/threads.h --env-main=env/main.c \
            --cc --cc-args=\"-llua5.3 -lpthread\"                           \
             --cc-output=/data/data/com.termux/files/usr/tmp/$(basename $i .ceu);                    \
	ceu --pre --pre-input=$i --pre-args=\"-I./include\"                \
        --ceu --ceu-features-async=true --ceu-features-os=true --ceu-features-lua=true --ceu-features-thread=true --ceu-features-dynamic=true --ceu-err-unused=pass \
	    --env --env-types=env/types.h --env-threads=env/threads.h --env-main=env/main.c \
            --cc --cc-args="-llua5.3 -lpthread"                             \
             --cc-output=/data/data/com.termux/files/usr/tmp/$(basename $i .ceu);                    \
	/data/data/com.termux/files/usr/tmp/$(basename $i .ceu);	                                        \
	echo ">>> OK";                                                      \
	echo;                                                               \
	echo;                                                               \
	echo;                                                               \
	echo;                                                               \
	echo;                                                               \
	echo;                                                               \
done

##########################################################################
File: samples/test-00.ceu -> /data/data/com.termux/files/usr/tmp/test-00
Description: Terminates immediatelly with an `escape`.
Features:
 - `escape`
##########################################################################
Press <enter> to start...ceu --pre --pre-input=samples/test-00.ceu --pre-args="-I./include" --ceu --ceu-features-os=true --ceu-features-lua=true --ceu-features-thread=true --ceu-features-dynamic=true --ceu-err-unused=pass --env --env-types=env/types.h --env-threads=env/threads.h --env-main=env/main.c --cc --cc-args="-llua5.3 -lpthread" --cc-output=/data/data/com.termux/files/usr/tmp/test-00

>>> OK







##########################################################################
File: samples/test-01.ceu -> /data/data/com.termux/files/usr/tmp/test-01
Description: Prints "Hello World" every second.
Features:
 - `every`  to react to time
 - `emit`   to simulate the passage of time
 - `async`  to be able to call `emit`
 - `par/or` to terminate after the `async` simulation
##########################################################################
Press <enter> to start...ceu --pre --pre-input=samples/test-01.ceu --pre-args="-I./include" --ceu --ceu-features-os=true --ceu-features-lua=true --ceu-features-thread=true --ceu-features-dynamic=true --ceu-err-unused=pass --env --env-types=env/types.h --env-threads=env/threads.h --env-main=env/main.c --cc --cc-args="-llua5.3 -lpthread" --cc-output=/data/data/com.termux/files/usr/tmp/test-01

Hello World!
Hello World!
Hello World!
Hello World!
Hello World!
>>> OK







##########################################################################
File: samples/test-02.ceu -> /data/data/com.termux/files/usr/tmp/test-02
Description: Prints "Hello World" every second.
Features:
 - same as `test-01.ceu`
 - `loop`   to simulate time as steps of `emit 1s`
 - `_sleep` to respect the time from the world
##########################################################################
Press <enter> to start...ceu --pre --pre-input=samples/test-02.ceu --pre-args="-I./include" --ceu --ceu-features-os=true --ceu-features-lua=true --ceu-features-thread=true --ceu-features-dynamic=true --ceu-err-unused=pass --env --env-types=env/types.h --env-threads=env/threads.h --env-main=env/main.c --cc --cc-args="-llua5.3 -lpthread" --cc-output=/data/data/com.termux/files/usr/tmp/test-02

Hello World!
Hello World!
Hello World!
Hello World!
Hello World!
>>> OK







##########################################################################
File: samples/test-03.ceu -> /data/data/com.termux/files/usr/tmp/test-03
Description: Interleaves "[sync] hello" and "[thread] world".
Features:
 - same as `test-02.ceu`
 - `async/thread` to execute an independent block in another OS thread
##########################################################################
Press <enter> to start...ceu --pre --pre-input=samples/test-03.ceu --pre-args="-I./include" --ceu --ceu-features-os=true --ceu-features-lua=true --ceu-features-thread=true --ceu-features-dynamic=true --ceu-err-unused=pass --env --env-types=env/types.h --env-threads=env/threads.h --env-main=env/main.c --cc --cc-args="-llua5.3 -lpthread" --cc-output=/data/data/com.termux/files/usr/tmp/test-03
>>> ERROR : samples/test-03.ceu:9:9: warning: implicit declaration of function 'pthread_cancel' is invalid in C99 [-Wimplicit-function-declaration]
        CEU_THREADS_CANCEL((CEU_APP.root.__thread_37)->id);
        ^
/data/data/com.termux/files/usr/tmp/lua_8c8m9v:71:51: note: expanded from macro 'CEU_THREADS_CANCEL'
#define CEU_THREADS_CANCEL(t)       ceu_assert_ex(pthread_cancel(t)==0, "bug found", CEU_TRACE_null)
                                                  ^
1 warning generated.
ld.lld: error: undefined symbol: pthread_cancel
>>> referenced by lua_8c8m9v
>>>               /data/data/com.termux/files/usr/tmp/lua_8c8m9v-f7bac7.o:(ceu_lbl)
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)

/data/data/com.termux/files/usr/bin/sh: 21: /data/data/com.termux/files/usr/tmp/test-03: not found
>>> OK







##########################################################################
File: samples/test-04.ceu -> /data/data/com.termux/files/usr/tmp/test-04
Description: Prints "Lua X" every second from three `lua` states.
Features:
 - three `lua` states: one global and two local
##########################################################################
Press <enter> to start...ceu --pre --pre-input=samples/test-04.ceu --pre-args="-I./include" --ceu --ceu-features-os=true --ceu-features-lua=true --ceu-features-thread=true --ceu-features-dynamic=true --ceu-err-unused=pass --env --env-types=env/types.h --env-threads=env/threads.h --env-main=env/main.c --cc --cc-args="-llua5.3 -lpthread" --cc-output=/data/data/com.termux/files/usr/tmp/test-04

Lua 1	0	0
Lua 2	0	0
Lua 3	0	0
Lua 1	1	1
Lua 2	1	1
Lua 3	1	1
Lua 1	2	2
Lua 2	2	2
Lua 3	2	2
Lua 1	3	3
Lua 2	3	3
Lua 3	3	3
Lua 1	4	4
Lua 2	4	4
Lua 3	4	4
>>> OK






```

**Comment by Grimler91** [comment]
Thanks!

**Pull request merged**